### PR TITLE
feat(web): include promoted results for DataSearch, CategorySearch, ReactiveComponent, ReactiveList and StateProvider

### PR DESCRIPTION
**Pull request by ShahAnuj2610**
dependent on: https://github.com/appbaseio/reactivecore/pull/55

### CODE REVIEW

File: packages/web/src/components/result/ReactiveList.js [hunk 1]
```diff
@@ -2,26 +2,27 @@ import React, { Component } from 'react';
...
-	getQueryOptions,
-	pushToAndClause,
-	getClassName,
-	parseHits,
 	checkSomePropChange,
-	getOptionsFromQuery,
+	getClassName,
 	getCompositeAggsQuery,
+	getOptionsFromQuery,
+	getQueryOptions,
```
Comment (bietkul) [inline packages/web/src/components/result/ReactiveList.js]: @ShahAnuj2610 Why there are formatting changes?

**Review by bietkul — CHANGES_REQUESTED** [review]
@ShahAnuj2610 Just make sure it works well with DataSearch, also make the same changes in `CategorySearch`.

File: packages/web/src/components/result/ReactiveList.js (hunk 1 above)
Comment (ShahAnuj2610) [reply packages/web/src/components/result/ReactiveList.js]: rearranged imports according to `alphabetical` order

**Review by bietkul — CHANGES_REQUESTED** [review]
@ShahAnuj2610 Provide `promotedResults` & `resultStats` in `render` method of search components.

File: packages/web/src/components/search/DataSearch.js [hunk 2]
```diff
@@ -5,45 +5,46 @@ import { withTheme } from 'emotion-theming';
...
-	setQueryOptions,
-	setQueryListener,
 	setComponentProps,
-	updateComponentProps,
+	setQueryListener,
+	setQueryOptions,
 	setSuggestionsSearchValue,
+	updateComponentProps,
+	updateQuery,
+	watchComponent,
```
Comment (bietkul) [inline packages/web/src/components/search/DataSearch.js]: @ShahAnuj2610 Remove these unnecessary changes, we talked about it.

**Review by bietkul — CHANGES_REQUESTED** [review]

File: packages/web/src/components/search/DataSearch.js (hunk 2 above)
Comment (ShahAnuj2610) [reply packages/web/src/components/search/DataSearch.js]: done

File: packages/web/src/components/search/DataSearch.js [hunk 3]
```diff
@@ -333,25 +334,7 @@ class DataSearch extends Component {
...
-			this.state.currentValue.toLowerCase(),
-		);
-
-		if (parseSuggestion) {
-			return parsedSuggestions.map(suggestion => parseSuggestion(suggestion));
-		}
-
-		return parsedSuggestions;
-	};
+	onSuggestions = results => handleOnSuggestions(results, this.state.currentValue, this.props);
```
Comment (bietkul) [inline packages/web/src/components/search/DataSearch.js:337]: @ShahAnuj2610 This is not what we talked about, avoid passing `this.props` in functions.
Don't change the `parseSuggestion` logic here, just do the sorting part for promoted results in `getSuggestions` method itself.
We don't need any new method here.

File: packages/web/src/components/search/CategorySearch.js [hunk 4]
```diff
@@ -410,22 +410,9 @@ class CategorySearch extends Component {
 		];
 	};
 
-	onSuggestions = (searchSuggestions) => {
-		const { parseSuggestion } = this.props;
-		const fields = Array.isArray(this.props.dataField)
-			? this.props.dataField
-			: [this.props.dataField];
+	onSuggestions = searchResults =>
```
Comment (bietkul) [inline packages/web/src/components/search/CategorySearch.js:413]: Make the same changes here too.

File: packages/web/src/components/basic/StateProvider.js [hunk 5]
```diff
@@ -95,6 +95,7 @@ const mapStateToProps = (state, props) => ({
 	aggregations: filterByComponentIds(state.aggregations, props),
 	isLoading: filterByComponentIds(state.isLoading, props),
 	error: filterByComponentIds(state.error, props),
+	promotedResults: filterByComponentIds(state.promotedResults, props),
 });
```
Comment (bietkul) [inline packages/web/src/components/basic/StateProvider.js:109]: @ShahAnuj2610 You didn't define the prop types here.

File: packages/web/src/components/basic/ReactiveComponent.js [hunk 6]
```diff
@@ -206,13 +210,28 @@ class ReactiveComponent extends Component {
...
+			}
+			filteredResults = [...promotedResults, ...filteredResults];
+		}
 		return {
-			data: parseHits(hits),
+			data: filteredResults,
 			aggregationData,
 			rawData: hits,
 			aggregations,
+			resultStats: this.stats,
```
Comment (bietkul) [inline packages/web/src/components/basic/ReactiveComponent.js:235]: @ShahAnuj2610 `promotedResults` is missing here

**Review by bietkul — CHANGES_REQUESTED** [review]
@ShahAnuj2610 Make the above changes

**Review by bietkul — APPROVED** [review]
